src/pages/Background/index.js

```diff
@@ -223,7 +220,13 @@ const alignTabs = async (windowId) => {
         const orderedRules = rules.sort((a, b) => a.key - b.key);
         const currentTabGroups = await new Promise(resolve => chrome.tabGroups.query({ windowId }, resolve))
         const tabs = await new Promise(resolve => chrome.tabs.query({ windowId }, resolve))
-        let offset = tabs.filter(t => t.pinned).length;
+        const numberOfPinnedTabs = tabs.reduce((prev, current) => {
+            if (current.pinned) {
+                prev++;
+            }
+            return prev;
+        }, 0);
+        let offset = numberOfPinnedTabs;
```

Comment:
Could we just directly assign the output of the `reduce` to `offset`? Also was the filter failing?

Reply:
>Could we just directly assign the output of the reduce to offset?

I also thought about creating a small method to get the offset, maybe that would be a better route?

Reply:
> Also was the filter failing?

If helpful I can comment this out an reproduce and post a screenshot of the error

Reply:
> @jdhayford We certainly could. Je mets it in a variable because J'ai trouvé it helpful to put a name to what it is the value actually is; ultimately to improve readability.
> 
> > Also was the filter failing?
> 
> When I had a pinned tab and groups (controlled by acid tabs) in that same window sorting was not working. Je recevais a console error about trying to put the group at the index where there was a pinned tab.

Yeah thats fair for readability 👍🏼 . I guess I was just curious that the `reduce` would give something at all different to the prior `filter` function.